### ISSUE TITLE
Support adding a specific field in Delta Lake connector

### ISSUE THREAD

**Issue by ebyhr** [trinodb/trino#16662]
Spark supports adding a field. #16248 is going to support it for Iceberg in Trino. It would be nice to support in Delta Lake connector as well.

**Comment by krvikash** [comment]
Hi @ebyhr, Could you please clarify if this issue is to support `specific field in nested field` as you linked https://github.com/trinodb/trino/issues/16248?

**Comment by ebyhr** [comment]
Right. However, the syntax needs review as we talked in Slack. Please wait for a while.
